### ISSUE TITLE
Cannot read property 'find' of undefined - Exit Handlers in  WorkflowTemplate

### ISSUE THREAD

**Issue by willgleich** [argoproj/argo#4061]
## Summary 

UI Error when viewing yaml section of exit handler steps that originate from a WorkflowTemplate. We tracked this down while investigating another UI bug that I believe(d?) to be related. But this one I was able to quickly reproduce locally by running a workflow --from workflowtemplate with a onExit dag template. I wonder if there is a way to improve these error messages into something more actionable. 

```apiVersion: argoproj.io/v1alpha1
kind: WorkflowTemplate
metadata:
  name: exit-handlers
spec:
  entrypoint: send-email
  onExit: exit-handler                  # invoke exit-hander template at end of the workflow
  templates:
  # primary workflow template
  - name: intentional-fail
    inputs:
      parameters:
      - name: about
        value: whales
    container:
      image: alpine:latest
      command: [sh, -c]
      args: ["echo intentional failure; exit 1"]
  # Exit handler templates
  # After the completion of the entrypoint template, the status of the
  # workflow is made available in the global variable {{workflow.status}}.
  # {{workflow.status}} will be one of: Succeeded, Failed, Error
  - name: exit-handler
    dag:
      tasks:
      - name: fail
        when: "Succeeded == Succeeded"
        continueOn:
          failed: true
          error: true
        template: intentional-fail
      - name: celebrate
        dependencies: [fail]
        template: celebrate
        when: "{{tasks.fail.status}} == Succeeded"
      - name: cry
        dependencies: [fail]
        template: cry
        when: "{{tasks.fail.status}} != Succeeded"
  - name: send-email
    container:
      image: alpine:latest
      command: [sh, -c]
      args: ["echo send e-mail: hello"]
  - name: celebrate
    container:
      image: alpine:latest
      command: [sh, -c]
      args: ["echo hooray!"]
  - name: cry
    container:
      image: alpine:latest
      command: [sh, -c]
      args: ["echo boohoo!"]
```
Then when I view the final failed step and click on YAML in the UI I see:
![image](https://user-images.githubusercontent.com/22464726/93517037-1fedca00-f8e8-11ea-8d8d-6683b636ccbf.png)
JS Console Error:
```
react-dom.production.min.js:198 TypeError: Cannot read property 'x' of undefined
    at workflow-dag.tsx:439
    at Array.forEach (<anonymous>)
    at t.getGraphSize (workflow-dag.tsx:438)
    at t.layoutGraphPretty (workflow-dag.tsx:418)
    at t.layoutGraph (workflow-dag.tsx:574)
    at t.render (workflow-dag.tsx:190)
    at Ei (react-dom.production.min.js:173)
    at Oi (react-dom.production.min.js:172)
    at Pi (react-dom.production.min.js:180)
    at Ua (react-dom.production.min.js:232)
```
FWIW The error message that led me to this presents its self by clicking on the node view and you are greeted with the below error message, I am still trying to replicate this in a concise way but I believe it to be related to exitHandlers as well.

```
 Cannot read property 'x' of undefined
 Reload this page to try again.

Stack Trace
TypeError: Cannot read property 'x' of undefined
    at https://argo.example.com/main.a0019f4931ab207b78a1.js:478:440151
    at Array.forEach (<anonymous>)
    at t.getGraphSize (https://argo.example.com/main.a0019f4931ab207b78a1.js:478:440118)
    at t.layoutGraphPretty (https://argo.example.com/main.a0019f4931ab207b78a1.js:478:439783)
    at t.layoutGraph (https://argo.example.com/main.a0019f4931ab207b78a1.js:478:442973)
    at t.render (https://argo.example.com/main.a0019f4931ab207b78a1.js:478:435352)
    at Ei (https://argo.example.com/main.a0019f4931ab207b78a1.js:396:62906)
    at Oi (https://argo.example.com/main.a0019f4931ab207b78a1.js:396:62701)
    at Pi (https://argo.example.com/main.a0019f4931ab207b78a1.js:396:66535)
    at Ua (https://argo.example.com/main.a0019f4931ab207b78a1.js:396:90552)
Component Stack

    in t
    in t
    in div
    in div
    in div
    in div
    in div
    in Unknown
    in t
    in t
    in t
    in Unknown
    in t
    in t
    in t
    in div
    in Unknown
    in t
    in n
```
JS Console Error:
```
TypeError: Cannot read property 'x' of undefined
    at workflow-dag.tsx:439
    at Array.forEach (<anonymous>)
    at t.getGraphSize (workflow-dag.tsx:438)
    at t.layoutGraphPretty (workflow-dag.tsx:418)
    at t.layoutGraph (workflow-dag.tsx:574)
    at t.render (workflow-dag.tsx:190)
    at Ei (react-dom.production.min.js:173)
    at Oi (react-dom.production.min.js:172)
    at Pi (react-dom.production.min.js:180)
    at Ua (react-dom.production.min.js:232)
```

## Diagnostics

What version of Argo Workflows are you running?
2.11.0-rc3
```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  creationTimestamp: "2020-09-17T16:23:44Z"
  generateName: exit-handlers-
  generation: 11
  labels:
    workflows.argoproj.io/completed: "true"
    workflows.argoproj.io/phase: Failed
    workflows.argoproj.io/workflow-template: exit-handlers
  name: exit-handlers-vx2xf
  namespace: dev
  resourceVersion: "1305502"
  selfLink: /apis/argoproj.io/v1alpha1/namespaces/dev/workflows/exit-handlers-vx2xf
  uid: 4bbff369-bca4-463a-bd90-c7a1a77545a4
spec:
  arguments: {}
  workflowTemplateRef:
    name: exit-handlers
status:
  conditions:
  - status: "True"
    type: Completed
  finishedAt: "2020-09-17T16:24:15Z"
  nodes:
    exit-handlers-vx2xf:
      displayName: exit-handlers-vx2xf
      finishedAt: "2020-09-17T16:24:03Z"
      hostNodeName: docker-desktop
      id: exit-handlers-vx2xf
      name: exit-handlers-vx2xf
      outputs:
        exitCode: "0"
      phase: Succeeded
      resourcesDuration:
        cpu: 3
        memory: 3
      startedAt: "2020-09-17T16:23:44Z"
      templateRef:
        name: exit-handlers
        template: send-email
      templateScope: local/
      type: Pod
    exit-handlers-vx2xf-654709357:
      boundaryID: exit-handlers-vx2xf-3947721577
      children:
      - exit-handlers-vx2xf-3142636042
      - exit-handlers-vx2xf-1581978203
      displayName: fail
      finishedAt: "2020-09-17T16:24:08Z"
      hostNodeName: docker-desktop
      id: exit-handlers-vx2xf-654709357
      inputs:
        parameters:
        - name: about
          value: whales
      message: failed with exit code 1
      name: exit-handlers-vx2xf.onExit.fail
      outputs:
        exitCode: "1"
      phase: Failed
      resourcesDuration:
        cpu: 2
        memory: 2
      startedAt: "2020-09-17T16:24:05Z"
      templateName: intentional-fail
      templateScope: local/
      type: Pod
    exit-handlers-vx2xf-1581978203:
      boundaryID: exit-handlers-vx2xf-3947721577
      displayName: cry
      finishedAt: "2020-09-17T16:24:13Z"
      hostNodeName: docker-desktop
      id: exit-handlers-vx2xf-1581978203
      name: exit-handlers-vx2xf.onExit.cry
      outputs:
        exitCode: "0"
      phase: Succeeded
      resourcesDuration:
        cpu: 2
        memory: 2
      startedAt: "2020-09-17T16:24:10Z"
      templateName: cry
      templateScope: local/
      type: Pod
    exit-handlers-vx2xf-3142636042:
      boundaryID: exit-handlers-vx2xf-3947721577
      displayName: celebrate
      finishedAt: "2020-09-17T16:24:10Z"
      id: exit-handlers-vx2xf-3142636042
      message: when 'Failed == Succeeded' evaluated false
      name: exit-handlers-vx2xf.onExit.celebrate
      phase: Skipped
      startedAt: "2020-09-17T16:24:10Z"
      templateName: celebrate
      templateScope: local/
      type: Skipped
    exit-handlers-vx2xf-3947721577:
      children:
      - exit-handlers-vx2xf-654709357
      displayName: exit-handlers-vx2xf.onExit
      finishedAt: "2020-09-17T16:24:15Z"
      id: exit-handlers-vx2xf-3947721577
      name: exit-handlers-vx2xf.onExit
      outboundNodes:
      - exit-handlers-vx2xf-3142636042
      - exit-handlers-vx2xf-1581978203
      phase: Failed
      startedAt: "2020-09-17T16:24:05Z"
      templateName: exit-handler
      templateScope: local/
      type: DAG
  phase: Failed
  resourcesDuration:
    cpu: 7
    memory: 7
  startedAt: "2020-09-17T16:23:44Z"
  storedTemplates:
    namespaced/exit-handlers/send-email:
      arguments: {}
      container:
        args:
        - 'echo send e-mail: hello'
        command:
        - sh
        - -c
        image: alpine:latest
        name: ""
        resources: {}
      inputs: {}
      metadata: {}
      name: send-email
      outputs: {}
  storedWorkflowTemplateSpec:
    arguments: {}
    entrypoint: send-email
    onExit: exit-handler
    templates:
    - arguments: {}
      container:
        args:
        - echo intentional failure; exit 1
        command:
        - sh
        - -c
        image: alpine:latest
        name: ""
        resources: {}
      inputs:
        parameters:
        - name: about
          value: whales
      metadata: {}
      name: intentional-fail
      outputs: {}
    - arguments: {}
      dag:
        tasks:
        - arguments: {}
          continueOn:
            error: true
            failed: true
          name: fail
          template: intentional-fail
          when: Succeeded == Succeeded
        - arguments: {}
          dependencies:
          - fail
          name: celebrate
          template: celebrate
          when: '{{tasks.fail.status}} == Succeeded'
        - arguments: {}
          dependencies:
          - fail
          name: cry
          template: cry
          when: '{{tasks.fail.status}} != Succeeded'
      inputs: {}
      metadata: {}
      name: exit-handler
      outputs: {}
    - arguments: {}
      container:
        args:
        - 'echo send e-mail: hello'
        command:
        - sh
        - -c
        image: alpine:latest
        name: ""
        resources: {}
      inputs: {}
      metadata: {}
      name: send-email
      outputs: {}
    - arguments: {}
      container:
        args:
        - echo hooray!
        command:
        - sh
        - -c
        image: alpine:latest
        name: ""
        resources: {}
      inputs: {}
      metadata: {}
      name: celebrate
      outputs: {}
    - arguments: {}
      container:
        args:
        - echo boohoo!
        command:
        - sh
        - -c
        image: alpine:latest
        name: ""
        resources: {}
      inputs: {}
      metadata: {}
      name: cry
      outputs: {}
```

```
time="2020-09-17T16:23:44Z" level=info msg="Processing workflow" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:23:44Z" level=info msg="Updated phase  -> Running" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:23:44Z" level=info msg="Pod node exit-handlers-vx2xf initialized Pending" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:23:44Z" level=info msg="Created pod: exit-handlers-vx2xf (exit-handlers-vx2xf)" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:23:44Z" level=info msg="Workflow update successful" namespace=dev phase=Running resourceVersion=1305395 workflow=exit-handlers-vx2xf
time="2020-09-17T16:23:45Z" level=info msg="Processing workflow" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:23:45Z" level=info msg="Updating node exit-handlers-vx2xf message: ContainerCreating"
time="2020-09-17T16:23:45Z" level=info msg="Skipped pod exit-handlers-vx2xf (exit-handlers-vx2xf) creation: already exists" namespace=dev podPhase=Pending workflow=exit-handlers-vx2xf
time="2020-09-17T16:23:45Z" level=info msg="Workflow update successful" namespace=dev phase=Running resourceVersion=1305400 workflow=exit-handlers-vx2xf
time="2020-09-17T16:23:46Z" level=info msg="Processing workflow" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:23:46Z" level=info msg="Skipped pod exit-handlers-vx2xf (exit-handlers-vx2xf) creation: already exists" namespace=dev podPhase=Pending workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:03Z" level=info msg="insignificant pod change" key=dev/exit-handlers-vx2xf
time="2020-09-17T16:24:04Z" level=info msg="Processing workflow" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:04Z" level=info msg="Setting node exit-handlers-vx2xf outputs"
time="2020-09-17T16:24:04Z" level=info msg="Updating node exit-handlers-vx2xf status Pending -> Running"
time="2020-09-17T16:24:04Z" level=info msg="Workflow update successful" namespace=dev phase=Running resourceVersion=1305443 workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:05Z" level=info msg="Processing workflow" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:05Z" level=info msg="Updating node exit-handlers-vx2xf status Running -> Succeeded"
time="2020-09-17T16:24:05Z" level=info msg="Running OnExit handler: exit-handler" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:05Z" level=info msg="DAG node exit-handlers-vx2xf-3947721577 initialized Running" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:05Z" level=info msg="All of node exit-handlers-vx2xf.onExit.fail dependencies [] completed" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:05Z" level=info msg="Pod node exit-handlers-vx2xf-654709357 initialized Pending" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:05Z" level=info msg="Created pod: exit-handlers-vx2xf.onExit.fail (exit-handlers-vx2xf-654709357)" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:05Z" level=info msg="Workflow update successful" namespace=dev phase=Running resourceVersion=1305448 workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:06Z" level=info msg="Processing workflow" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:06Z" level=info msg="Updating node exit-handlers-vx2xf-654709357 message: ContainerCreating"
time="2020-09-17T16:24:06Z" level=info msg="Labeled pod completed" namespace=dev pod=exit-handlers-vx2xf
time="2020-09-17T16:24:06Z" level=info msg="Running OnExit handler: exit-handler" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:06Z" level=info msg="Skipped pod exit-handlers-vx2xf.onExit.fail (exit-handlers-vx2xf-654709357) creation: already exists" namespace=dev podPhase=Pending workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:06Z" level=info msg="Workflow update successful" namespace=dev phase=Running resourceVersion=1305458 workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:07Z" level=info msg="Processing workflow" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:07Z" level=info msg="Running OnExit handler: exit-handler" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:07Z" level=info msg="Labeled pod completed" namespace=dev pod=exit-handlers-vx2xf
time="2020-09-17T16:24:07Z" level=info msg="Skipped pod exit-handlers-vx2xf.onExit.fail (exit-handlers-vx2xf-654709357) creation: already exists" namespace=dev podPhase=Pending workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:08Z" level=info msg="insignificant pod change" key=dev/exit-handlers-vx2xf-654709357
time="2020-09-17T16:24:09Z" level=info msg="Processing workflow" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:09Z" level=info msg="Setting node exit-handlers-vx2xf-654709357 outputs"
time="2020-09-17T16:24:09Z" level=info msg="Updating node exit-handlers-vx2xf-654709357 status Pending -> Running"
time="2020-09-17T16:24:09Z" level=info msg="Running OnExit handler: exit-handler" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:09Z" level=info msg="Workflow update successful" namespace=dev phase=Running resourceVersion=1305471 workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:10Z" level=info msg="Processing workflow" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:10Z" level=info msg="Updating node exit-handlers-vx2xf-654709357 status Running -> Failed"
time="2020-09-17T16:24:10Z" level=info msg="Updating node exit-handlers-vx2xf-654709357 message: failed with exit code 1"
time="2020-09-17T16:24:10Z" level=info msg="Running OnExit handler: exit-handler" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:10Z" level=info msg="All of node exit-handlers-vx2xf.onExit.celebrate dependencies [fail] completed" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:10Z" level=info msg="Skipped node exit-handlers-vx2xf-3142636042 initialized Skipped (message: when 'Failed == Succeeded' evaluated false)" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:10Z" level=info msg="All of node exit-handlers-vx2xf.onExit.cry dependencies [fail] completed" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:10Z" level=info msg="Pod node exit-handlers-vx2xf-1581978203 initialized Pending" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:10Z" level=info msg="Created pod: exit-handlers-vx2xf.onExit.cry (exit-handlers-vx2xf-1581978203)" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:10Z" level=info msg="Workflow update successful" namespace=dev phase=Running resourceVersion=1305478 workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:11Z" level=info msg="Processing workflow" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:11Z" level=info msg="Labeled pod completed" namespace=dev pod=exit-handlers-vx2xf-654709357
time="2020-09-17T16:24:11Z" level=info msg="Updating node exit-handlers-vx2xf-1581978203 message: ContainerCreating"
time="2020-09-17T16:24:11Z" level=info msg="Running OnExit handler: exit-handler" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:11Z" level=info msg="Skipped pod exit-handlers-vx2xf.onExit.cry (exit-handlers-vx2xf-1581978203) creation: already exists" namespace=dev podPhase=Pending workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:11Z" level=info msg="Workflow update successful" namespace=dev phase=Running resourceVersion=1305485 workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:12Z" level=info msg="Processing workflow" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:12Z" level=info msg="Running OnExit handler: exit-handler" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:12Z" level=info msg="Skipped pod exit-handlers-vx2xf.onExit.cry (exit-handlers-vx2xf-1581978203) creation: already exists" namespace=dev podPhase=Pending workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:13Z" level=info msg="insignificant pod change" key=dev/exit-handlers-vx2xf-1581978203
time="2020-09-17T16:24:14Z" level=info msg="Processing workflow" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:14Z" level=info msg="Setting node exit-handlers-vx2xf-1581978203 outputs"
time="2020-09-17T16:24:14Z" level=info msg="Updating node exit-handlers-vx2xf-1581978203 status Pending -> Running"
time="2020-09-17T16:24:14Z" level=info msg="Running OnExit handler: exit-handler" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:14Z" level=info msg="Workflow update successful" namespace=dev phase=Running resourceVersion=1305498 workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:15Z" level=info msg="Processing workflow" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:15Z" level=info msg="Updating node exit-handlers-vx2xf-1581978203 status Running -> Succeeded"
time="2020-09-17T16:24:15Z" level=info msg="Running OnExit handler: exit-handler" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:15Z" level=info msg="Outbound nodes of exit-handlers-vx2xf-3947721577 set to [exit-handlers-vx2xf-3142636042 exit-handlers-vx2xf-1581978203]" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:15Z" level=info msg="node exit-handlers-vx2xf-3947721577 phase Running -> Failed" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:15Z" level=info msg="node exit-handlers-vx2xf-3947721577 finished: 2020-09-17 16:24:15.8400411 +0000 UTC" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:15Z" level=info msg="Checking daemoned children of exit-handlers-vx2xf-3947721577" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:15Z" level=info msg="Updated phase Running -> Failed" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:15Z" level=info msg="Marking workflow completed" namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:15Z" level=info msg="Checking daemoned children of " namespace=dev workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:15Z" level=info msg="Workflow update successful" namespace=dev phase=Failed resourceVersion=1305502 workflow=exit-handlers-vx2xf
time="2020-09-17T16:24:16Z" level=info msg="Labeled pod completed" namespace=dev pod=exit-handlers-vx2xf-1581978203
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by alexec** [comment]
@simster7 fixed by #3666 ?

**Comment by simster7** [comment]
I have confirmed this is fixed on master. Fix should be on next release
